### PR TITLE
LINKTA-142: Setup Zustand

### PR DESCRIPTION
**Pull request by ejwoods**
## Description

- Added `navigationStore.ts` file to store directory to demonstrate how Zustand stores can be created and [establish best practices](https://tkdodo.eu/blog/working-with-zustand):
  - Only export custom hooks
  - Prefer atomic selectors
  - Separate actions from state
- Added basic demo of using navigationStore to `App.jsx`. (Know this file is being replaced in React Router PR - this is just for demo purposes if anyone would like to see Zustand in action right now)

## Type of change

- [ ]  🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨  New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] Other 🗒️ ______________

## How Has This Been Tested?

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I have reviewed the [**Security Best Practices Document**](SECURITY_BEST_PRACTICES.md).
- [x] I have followed the [**Naming Conventions Guide**](NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [x] I have ensured that my pull request title is descriptive.

### CODE REVIEW

File: client/App.jsx [hunk 1]
```diff
@@ -1,5 +1,25 @@
 import React from 'react';
+import {
+  useCurrentPage,
+  useNavigationActions,
+} from './store/navigationStore.ts';
 
 export default function App() {
-  return <h1>Linktaaaaaaaa</h1>;
```
Comment (ChTiSh) [inline client/App.jsx:4]: We probably need to remove the demo part to avoid the merge conflicts

**Comment by ChTiSh** [comment]
@ejwoods thank you so much for establishing the best practices for zustand and create a demo as well. Losing these would be a shame, do you think would it be possible to make it into a best practice documentation file along with the demo? I will create a seperate ticket for that.

I also add a description for the user input store setup under the ticket description as I realized it wouldn't be able to setup zustand without a use case. Sorry sorry

**Review by ChTiSh — CHANGES_REQUESTED** [review]

File: client/App.jsx (hunk 1 above)
Comment (ejwoods) [reply client/App.jsx:4]: It's my understanding that this `App.jsx` file will be removed entirely and replaced by `App.tsx` when Andrew's React Router PR is merged. I can delete this file when we are ready to merge if that works.

**Comment by ejwoods** [comment]
> @ejwoods thank you so much for establishing the best practices for zustand and create a demo as well. Losing these would be a shame, do you think would it be possible to make it into a best practice documentation file along with the demo? I will create a seperate ticket for that.
> 
> I also add a description for the user input store setup under the ticket description as I realized it wouldn't be able to setup zustand without a use case. Sorry sorry

I can make a best practice doc if you think it will be helpful - maybe we can discuss in standup what you would like that to look like. 

I am looking to start on the input components today and will also set up a Zustand store for that. This issue LINKTA-142 is under the navigation epic (LINKTA-52), so that's why I started by creating the navigation store. Should we create a separate issue for the input store under **LINKTA-40 User is able to submit tree prompt**?

File: client/store/navigationStore.ts
[the remark this answers is not in the document]
Comment (ejwoods) [reply client/store/navigationStore.ts:1]: Interesting - I will look into this.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 .env
 node_modules
+**/.DS_Store
```
Comment (ejwoods) [inline .gitignore:3]: I noticed that there is another pending PR that adds this to the gitignore as well so I can correct that conflict when we are ready to merge this PR.

**Comment by ChTiSh** [comment]
Yeah, we should have a separate ticket under the input epic. Let's discuss tomorrow.

**Pull request closed without merging**